distributed/cli/tests/test_dask_worker.py

```diff
@@ -149,24 +167,60 @@ def test_no_nanny(loop):
 
 @pytest.mark.slow
 @pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
-def test_no_reconnect(nanny, loop):
-    with popen(["dask-scheduler", "--no-dashboard"]) as sched:
-        wait_for_port(("127.0.0.1", 8786))
+@pytest.mark.asyncio
+async def test_no_reconnect(nanny):
+    async with Scheduler(dashboard_address=":0") as s, Client(
+        s.address, asynchronous=True
+    ) as c:
         with popen(
             [
                 "dask-worker",
-                "tcp://127.0.0.1:8786",
+                s.address,
                 "--no-reconnect",
                 nanny,
                 "--no-dashboard",
             ]
         ) as worker:
-            sleep(2)
-            terminate_process(sched)
```

Comment:
This is the only place that util was used, you can also delete `terminate_process` while you're at it.

Reply:
it's still used in `utils_test.popen` - so I made it private instead